src/dolphin/util.ts

```diff
@@ -77,20 +104,69 @@ export async function addGamePathToInis(gameDir: string): Promise<void> {
   const userFolder = await findUserFolder(type);
   const iniPath = path.join(userFolder, "Config", "Dolphin.ini");
+  const iniFile = new IniFile();
   if (await fileExists(iniPath)) {
     log.info("Found a Dolphin.ini to update...");
-    const dolphinINI = ini.parse(await fs.readFile(iniPath, "utf-8"));
-    dolphinINI.General.ISOPath0 = gameDir;
-    const numPaths = dolphinINI.General.ISOPaths;
-    dolphinINI.General.ISOPaths = numPaths !== "0" ? numPaths : "1";
-    const newINI = ini.encode(dolphinINI);
-    await fs.writeFile(iniPath, newINI);
+    await iniFile.load(iniPath, false);
+    const generalSection = iniFile.getOrCreateSection("General");
+    const numPaths = generalSection.get("ISOPaths", "0");
+    generalSection.set("ISOPaths", numPaths !== "0" ? numPaths : "1");
+    generalSection.set("ISOPath0", gameDir);
   } else {
     log.info("There isn't a Dolphin.ini to update...");
-    const configPath = path.join(userFolder, "Config");
-    const newINI = ini.encode({ General: { ISOPath0: gameDir, ISOPaths: 1 } });
-    await fs.ensureDir(configPath);
-    await fs.writeFile(iniPath, newINI);
+    const generalSection = iniFile.getOrCreateSection("General");
+    generalSection.set("ISOPaths", "1");
+    generalSection.set("ISOPath0", gameDir);
   }
+  iniFile.save(iniPath);
   log.info(`Finished updating ${type} dolphin...`);
 }
+
+export async function updateOnlineBoot(enable = false) {
+  const userPath = await findUserFolder(DolphinLaunchType.NETPLAY);
+  const sysPath = await findSysFolder(DolphinLaunchType.NETPLAY);
+  const globalIniPath = path.join(sysPath, "GameSettings", "GALE01r2.ini");
+  const localIniPath = path.join(userPath, "GameSettings", "GALE01.ini");
+  await bootToCSS(globalIniPath, localIniPath, enable);
+}
+
+async function bootToCSS(globalIniPath: string, localIniPath: string, enable: boolean) {
+  const globalIni = new IniFile();
+  const localIni = new IniFile();
+  if (await fs.pathExists(globalIniPath)) {
+    log.info(`found global ini file: ${globalIniPath}`);
+    await globalIni.load(globalIniPath);
+  }
+  if (await fs.pathExists(localIniPath)) {
+    log.info(`found local ini file: ${localIniPath}`);
+    await localIni.load(localIniPath);
+  }
+
+  const geckoCodes = loadGeckoCodes(globalIni, localIni);
+
+  log.info(geckoCodes);
+
+  const bootCodeIdx = geckoCodes.findIndex((code) => code.name === "Boot to CSS");
+
+  if (bootCodeIdx === -1) {
+    const bootToCssCode: GeckoCode = {
+      codeLines: ["041BFA20 38600002"],
+      creator: "Dan Salvato, Achilles",
+      defaultEnabled: false,
+      enabled: enable,
+      name: "Boot to CSS ",
+      notes: [],
+      userDefined: true,
+    };
+    geckoCodes.push(bootToCssCode);
+  } else {
+    if (geckoCodes[bootCodeIdx].enabled === enable) {
+      return;
+    }
+    geckoCodes[bootCodeIdx].enabled = enable;
+  }
+
+  saveCodes(localIni, geckoCodes);
+
+  localIni.save(localIniPath);
```

Comment:
How come we load `globalIni` and `localIni` in the beginning of this function, but when we call save, we only save the localIni?

Reply:
we load inis from Sys and User to have complete knowledge of what codes are enabled and how. We only write to the ini in User because otherwise we would be overwriting the defaults. This is standard behavior in Dolphin.